src/HL7/Segments/PD1.php

```diff
@@ -0,0 +1,231 @@
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setPatientPrimaryFacility($value, int $position = 3): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setPatientPrimaryCareProviderNameAndIDNo($value, int $position = 4): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setStudentIndicator($value, int $position = 5): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setHandicap($value, int $position = 6): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setLivingWillCode($value, int $position = 7): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setOrganDonorCode($value, int $position = 8): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setSeparateBill($value, int $position = 9): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setDuplicatePatient($value, int $position = 10): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setPublicityCode($value, int $position = 11): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setProtectionIndicator($value, int $position = 12): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setProtectionIndicatorEffectiveDate($value, int $position = 13): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setPlaceOfWorship($value, int $position = 14): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setAdvanceDirectiveCode($value, int $position = 15): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setImmunizationRegistryStatus($value, int $position = 16): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setImmunizationRegistryStatusEffectiveDate($value, int $position = 17): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setPublicityCodeEffectiveDate($value, int $position = 18): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setMilitaryBranch($value, int $position = 19): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setMilitaryRankGrade($value, int $position = 20): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function setMilitaryStatus($value, int $position = 21): bool
+    {
+        return $this->setField($position, $value);
+    }
+
+    public function getLivingDependency(int $position = 1): int|array|string|null
```

Comment:
Use specific return types instead of `int|array|string|null` for the get* methods, wherever applicable please. From the documentation it looks like all fields are optional, so `|null` can be added for all.

Reply:
Wouldn't it be too rigid? I mean, for example RXA.3 should be array since it is composed of two fields TS.1 and TS.2 (Time and Precision), however only Time is required. So the user can pass an array as well as a text.

I could leave it without return type as the rest of the segments if you like, but I don't know if it is a good idea to restrict it too much, considering that the sets are open.

Reply:
If it can return an array or a string use `array|string`, if empty value is a possibility use `array|string|null`. When we know for sure that a field won't return an integer, we shouldn't be adding `|int`. 

The idea is to restrict it just enough to prevent wrong types being returned. Strict type checking prevents a lot of hard-to-detect bugs down the line.